File: sample-projects/python/de/sots/cellarsandcentaurs/application/creature_util.py
from de.sots.cellarsandcentaurs.domain.model import *
from de.sots.cellarsandcentaurs.domain.model.fightable import Fightable


class CreatureUtil:
    STANDARD_ARMOR_CLASS_DESCRIPTION = "Natural Armor"

    @staticmethod
    def count_treasure_hoard(creature: Creature) -> int:
        """Counts the treasure hoard a creature guards."""
        return creature.xp_value * 10

    @staticmethod
    def is_wounded(hit_points: HitPoints) -> bool:
        return hit_points.current < hit_points.max

File: sample-projects/kotlin/src/main/kotlin/de/sots/cellarsandcentaurs/application/CreatureUtil.kt
package de.sots.cellarsandcentaurs.application

import de.sots.cellarsandcentaurs.domain.model.*
import de.sots.cellarsandcentaurs.domain.model.Fightable
import org.slf4j.Logger
import org.slf4j.LoggerFactory

object CreatureUtil {
    const val STANDARD_ARMOR_CLASS_DESCRIPTION: String = "Natural Armor"

    private val logger: Logger = LoggerFactory.getLogger(CreatureUtil::class.java)

    /*
     * Counts the treasure hoard a creature guards.
     */
    fun treasureHoardOf(creature: Creature): Int {
        val hitPoints = creature.hitPoints ?: HitPoints.init(1)
        logger.info("Counting hoard of {}", creature.id)
        return hitPoints.max * DEFAULT_DIE_SIDES
    }
}
